### DISCUSSION TITLE
对 OpenAI API 使用“结构化输出”进行“多段翻译”请求

### DISCUSSION THREAD

**Discussion by PositionZer0** [immersive-translate/immersive-translate#2063] [Ideas]
### 新功能 | New Feature

1. 为兼容 OpenAI API 的翻译服务添加选项，以使用“结构化输出”进行“多段翻译”请求

### 为什么您想要添加这些功能 | Why you want to add these features

1. 目前使用的 “多段翻译”偶尔会产生不符合 YAML 格式要求的响应
2. 使用“格式化输出”可以确保模型生成的响应始终符合 JSON 格式
3. `gpt-3.5-turbo`, `gpt-4-*`, `gpt-4o-*`, `deepseek-chat` 等模型已支持 JSON Output 功能

### 提议 | Proposal

需要修改原来 prompt 中的 YAML 为 JSON 格式。

参考 OpenAI API 和 DeepSeek API 的开发文档：
- https://platform.openai.com/docs/guides/structured-outputs/introduction
- https://platform.deepseek.com/api-docs/zh-cn/guides/json_mode

**Comment by theowenyoung** [comment]
感谢，我们研究一下。
